app/components/primer/alpha/action-menu-element.ts

```diff
@@ -8,39 +8,66 @@ export {}
 
 class ActionMenuElement extends HTMLElement {
   #abortController: AbortController
-  // eslint-disable-next-line no-invalid-this
-  #actionMenuEl: HTMLElement = this
-  // eslint-disable-next-line no-invalid-this
-  #buttonEl = this.querySelector<HTMLButtonElement>('button')
-  // eslint-disable-next-line no-invalid-this
-  #menuEl = this.querySelector<HTMLUListElement>('[role="menu"]')
-  #menuItemEls: HTMLElement[] = []
+  #firstCharactersOfItems: string[]
   #firstMenuItem: HTMLElement
   #lastMenuItem: HTMLElement
-  #firstCharactersOfItems: string[] = []
-  // eslint-disable-next-line no-undef
-  #allMenuItemEls: NodeListOf<HTMLElement> | null =
-    // eslint-disable-next-line no-invalid-this
-    this.#menuEl && this.#menuEl.querySelectorAll('[role="menuitem"],[role="menuitemcheckbox"],[role="menuitemradio"]')
 
   get anchorAlign(): AnchorAlignment {
     return (this.getAttribute('data-anchor-align') || 'start') as AnchorAlignment
   }
 
-  set anchorAlign(value: AnchorAlignment) {
-    this.setAttribute('data-anchor-align', value)
-  }
-
   get anchorSide(): AnchorSide {
     return (this.getAttribute('data-anchor-side') || 'outside-bottom') as AnchorSide
   }
 
-  set anchorSide(value: AnchorSide) {
-    this.setAttribute('data-anchor-align', value)
+  get menu(): HTMLUListElement | null {
```

Comment:
Thanks for doing this! I'm curious what this does as opposed to them being defined at the top?

Reply:
This makes sure that the variables are only grabbed when we need it, AFTER the element and its children are properly loaded by the browser. I think this relates to why we saw our event handlers weren't working with our documentation site in some cases before we added `<script defer>`

Reply:
Ah! Ok wonderful thank you